packages/x-charts/src/internals/components/ChartsAxesGradients/ChartsPiecewiseGradient.tsx

```diff
@@ -31,7 +31,7 @@ export default function ChartsPiecewiseGradient(props: ChartsPiecewiseGradientPr
         }
         const offset = isReversed ? 1 - x / size : x / size;
```

Comment:
The only reasons this could be infinit are:

1. `x` is inifinit
2. `size` is 0

If it's because of the second, we can ignore this gradient since it does not have size

Reply:
Nice catch